tests/asynchronous_fifo_asrc_test/src/asynchronous_fifo_asrc_test.c

```diff
@@ -1,58 +1,270 @@
-// Copyright 2023-2024 XMOS LIMITED.
+// Copyright 2024 XMOS LIMITED.
 // This Software is subject to the terms of the XMOS Public Licence: Version 1.
 
 #include <xcore/parallel.h>
 #include <xcore/hwtimer.h>
+#include <xs1.h>
+#include <stdlib.h>
 #include <stdio.h>
 #include <print.h>
 #include <platform.h>
 #include <xscope.h>
 #include "asynchronous_fifo.h"
+#include "asrc_timestamp_interpolation.h"
 
-DECLARE_JOB(producer, (asynchronous_fifo_t *));
-DECLARE_JOB(consumer, (asynchronous_fifo_t *));
+#include <string.h>
+
+#include "src.h"
+
+
+int fs_code(int frequency) {
```

Comment:
I see this is merged but I'd recommend this fn goes in the lib as it is usually needed in the app when initialising the SRC, although maybe with an assert rather than exit.